README.md

```diff
@@ -0,0 +1,546 @@
+# Coding Guidelines for NBIS Developers
+
+This document is is currently a work in progress, and presents practical
+guidelines for developers and others at NBIS who write or contribute to
+software. The guidelines cover:
+
+* coding style and documentation,
+* licensing, packaging and distribution of software,
+* how we use Git and GitHub in large and small projects, and
+* how we do code reviews.
+
+This is a reference document, and as such you are expected to dip into
+it to read the parts that you need to know something about.
+
+* If you only occasionally write code, or if you write code outside
+of the NBIS system developers’ team, you should read the full text of
+this document at least once, but there is no expectation that you will
+remember it all.
+* If you are a member of the NBIS system developers’ team, you should
+make yourself familiar with these guidelines.
+* In any case, these are *guidelines*, not hard rules. Apart from the
+restrictions placed upon NBIS by our funding agencies, we do not force
+these guidelines on you or on ourselves. We do however think that they
+would help to streamline our workflow, improve teamwork and benefit both
+ourselves individually as well as the "products" that we produce as
+developers.
+
+The NBIS team of system developers are more than happy to try to answer
+any questions you may have about the contents of this document. We are
+available on Slack. The "code-review-forum" channel is a good place to
+ask.
+
+Remember, these guidelines are supposed to make everything we produce
+better, so that we can say *"I was a part of that!"*. It is important
+that we don’t feel that we lose too much velocity trying to adhere to
+these guidelines. If you feel that they slow things down too much, we
+need to review them. These things are not set in stone.
+
+Table of Contents
+=================
+
+  * [Coding Guidelines for NBIS Developers](#coding-guidelines-for-nbis-developers)
+    * [Things to be aware of when writing code](#things-to-be-aware-of-when-writing-code)
+      * [Intent](#intent)
+      * [Comments in code](#comments-in-code)
+      * [Readability](#readability)
+      * [Best programming practices](#best-programming-practices)
+      * [Documentation and packaging](#documentation-and-packaging)
+        * [Files bundled with a piece of software](#files-bundled-with-a-piece-of-software)
+      * [Sensitive data](#sensitive-data)
+      * [Testing](#testing)
+    * [How we use GitHub](#how-we-use-github)
+    * [How we use Git](#how-we-use-git)
+    * [General stuff about working with Git](#general-stuff-about-working-with-git)
+    * [How we do code reviews](#how-we-do-code-reviews)
+      * [Code reviewing steps](#code-reviewing-steps)
+      * [General stuff about code reviews](#general-stuff-about-code-reviews)
+
+## Things to be aware of when writing code
+
```

Comment:
Converted this big list into subsections. I thought that the text was too long for a list.